File: CODIGO/AdoptaYA-WebAPI/src/models/permiso.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const permiso = sequelize.define('permiso', {
    id_rol:    { type: DataTypes.INTEGER.UNSIGNED, primaryKey: true },
    id_modulo: { type: DataTypes.INTEGER.UNSIGNED, primaryKey: true },
    create:    { type: DataTypes.TINYINT(1), allowNull: true },
    read:      { type: DataTypes.TINYINT(1), allowNull: true },
    update:    { type: DataTypes.TINYINT(1), allowNull: true },
    delete:    { type: DataTypes.TINYINT(1), allowNull: true }
  }, {
    tableName: 'permiso'
  });

  permiso.associate = (models) => {
    permiso.belongsTo(models.rol,    { foreignKey: 'id_rol' });
    permiso.belongsTo(models.modulo, { foreignKey: 'id_modulo' });
  };

  return permiso;
};

File: DOCUMENTOS/CODIGO/AdoptaYA/Functionalities/Permissions/Model/Permiso.cs
using AdoptaYA.Functionalities.Modules.Model;
using System.Text.Json.Serialization;

namespace AdoptaYA.Functionalities.Permissions.Model;

public class Permiso
{
    [JsonPropertyName("id_rol")]
    public int RoleId { get; set; }

    [JsonPropertyName("id_modulo")]
    public int ModuleId { get; set; }

    [JsonPropertyName("create")]
    public int Create { get; set; }

    [JsonPropertyName("read")]
    public int Read { get; set; }

    [JsonPropertyName("update")]
    public int Update { get; set; }

    [JsonPropertyName("delete")]
    public int Delete { get; set; }

    [JsonPropertyName("modulo")]
    public Modulo? Module { get; set; }
}